src/main/java/seedu/address/model/Model.java

```diff
@@ -19,14 +20,31 @@
     void resetData(ReadOnlyTaskBook newData);
 
     /** Returns the TaskBook */
-    ReadOnlyTaskBook getAddressBook();
+    ReadOnlyTaskBook getTaskBook();
 
     /** Deletes the given task. */
     void deleteTask(Task target) throws TaskNotFoundException;
 
     /** Adds the given task */
     void addTask(Task task);
 
+    //undo and redo
+    /** Undo prev action that modifies data**/
+    Command undo();
+
+    /** reset stack of redoable actions when a non undo modifying data command is called**/
+    void resetRedoables();
+
+    void recordState(Command command);
+
+    boolean hasUncommittedChanges();
+
+    void discardRecentCommit();
```

Comment:
At most, I only expect to see `hasUncommitedChanges()`, `undo()`, `redo()` and `recordState()`.